planner/core/memtable_predicate_extractor.go

```diff
@@ -458,6 +458,114 @@ func (helper extractHelper) extractTimeRange(
+		return ast.GT
+	case tipb.ScalarFuncSig_GEInt, tipb.ScalarFuncSig_GEReal, tipb.ScalarFuncSig_GEDecimal:
+		return ast.GE
+	case tipb.ScalarFuncSig_LTInt, tipb.ScalarFuncSig_LTReal, tipb.ScalarFuncSig_LTDecimal:
+		return ast.LT
+	case tipb.ScalarFuncSig_LEInt, tipb.ScalarFuncSig_LEReal, tipb.ScalarFuncSig_LEDecimal:
+		return ast.LE
+	default:
+		return fn.FuncName.L
+	}
+}
+
+// Extracts the time range column from `TiDB_Hot_Regions_History`, it use function pb code to
+// distinguish from Int and (Real,Decimal) in GT and LE function.
+// It leave EQ function to above SELECTION excutor different from getTimeFunctionName.
+// Currently, this is used by `TiDB_Hot_Regions_History` only.
+// e.g: SELECT * FROM t WHERE a>90 AND a<100
+func (helper extractHelper) extractQuantilesRange(
+	ctx sessionctx.Context,
+	schema *expression.Schema,
+	names []*types.FieldName,
+	predicates []expression.Expression,
+	extractColName string,
+) (
+	remained []expression.Expression,
+	// unix timestamp in nanoseconds
+	lowQuantile float64,
+	HighQuantile float64,
+) {
+	var smallFloat64 = 1e-12
+	var smallInt64 = float64(1)
+	remained = make([]expression.Expression, 0, len(predicates))
+	extractCols := helper.findColumn(schema, names, extractColName)
+	if len(extractCols) == 0 {
+		return predicates, lowQuantile, HighQuantile
+	}
+
+	for _, expr := range predicates {
+		fn, ok := expr.(*expression.ScalarFunction)
+		if !ok {
+			remained = append(remained, expr)
+			continue
+		}
+
+		var colName string
+		var datums []types.Datum
+		fnName := helper.getQuantilesFunctionName(fn)
+		switch fnName {
+		case ast.GT, ast.GE, ast.LT, ast.LE:
+			colName, datums = helper.extractColBinaryOpConsExpr(extractCols, fn)
+		}
+
+		if colName == extractColName {
+			doubleType := types.NewFieldType(mysql.TypeDouble)
+			doubleType.Decimal = 12
+			doubleDatum, err := datums[0].ConvertTo(ctx.GetSessionVars().StmtCtx, doubleType)
+			if err != nil || doubleDatum.Kind() == types.KindNull {
+				remained = append(remained, expr)
+				continue
+			}
+			mysqlDouble := doubleDatum.GetFloat64()
+
+			switch fnName {
+			case ast.GT:
+				// Fixme: add a samll number is not absolutely correct here
+				// add 1e-12 for float 64, add 1 for int
+				if fn.Function.PbCode() == tipb.ScalarFuncSig_GTInt {
+					lowQuantile = math.Max(lowQuantile, mysqlDouble+smallInt64)
+				} else {
+					lowQuantile = math.Max(lowQuantile, mysqlDouble+smallFloat64)
+				}
+			case ast.GE:
+				lowQuantile = math.Max(lowQuantile, mysqlDouble)
+			case ast.LT:
+				if HighQuantile == 0 {
+					if fn.Function.PbCode() == tipb.ScalarFuncSig_LTInt {
+						HighQuantile = mysqlDouble - smallInt64
+					} else {
+						HighQuantile = mysqlDouble - smallFloat64
+					}
+				} else {
+					if fn.Function.PbCode() == tipb.ScalarFuncSig_LTInt {
+						HighQuantile = math.Min(HighQuantile, HighQuantile-smallInt64)
+					} else {
+						HighQuantile = math.Min(HighQuantile, HighQuantile-smallFloat64)
```

Comment:
`extractQuantilesRange` is added to push down `hot_degree`,` flow_bytes`, `key_rate`, `query_rate` is strange in case of `GT`,`LT`.  We can remove this function,If there is no need to push down these fields.

Reply:
I have removed it.